lib/concurrent/edge/future.rb

```diff
@@ -100,25 +128,26 @@ def initialize(promise, default_executor = :io)
         ensure_ivar_visibility!
       end
 
+      # @return [:pending, :completed]
       def state
-        @State.get
+        @State.get.to_sym
```

Comment:
Ah ok, so if I understand correctly this will only return either `:success` or `:failed`?